azurecompute-arm/src/test/java/org/jclouds/azurecompute/arm/features/DeploymentApiLiveTest.java

```diff
@@ -0,0 +1,129 @@
+              "    \"template\" : " + template + ", " +
+              "    \"mode\" : \"" + mode + "\", " +
+              "    \"parameters\" : " + parameters + " " +
+              "  } " +
+              "}";
+      return body;
+   }
+
+   @Test(groups = "live")
+   public void testCreate() {
+      Deployment deployment = api().create(deploymentName, properties);
+      assertNotNull(deployment);
+
+      //Poll until resource is ready to be used
+      boolean jobDone = Predicates2.retry(new Predicate<String>() {
+         @Override
+         public boolean apply(String name) {
+            Deployment dp = api().get(deploymentName);
+            ProvisioningState state = ProvisioningState.fromValue(dp.properties().provisioningState());
+            return state == ProvisioningState.SUCCEEDED;
+         }
+      }, 60 * maxTestDuration * 1000).apply(deploymentName);
+      assertTrue(jobDone, "create operation did not complete in the configured timeout");
+
+      Deployment dp = api().get(deploymentName);
+      ProvisioningState state = ProvisioningState.fromValue(dp.properties().provisioningState());
+      assertTrue(state == ProvisioningState.SUCCEEDED);
+   }
+
+
+   @Test(groups = "live", dependsOnMethods = "testCreate")
+   public void testGetDeployment() {
+      Deployment deployment = api().get(deploymentName);
+      assertNotNull(deployment);
+      ProvisioningState state = ProvisioningState.fromValue(deployment.properties().provisioningState());
+      assertTrue(state == ProvisioningState.SUCCEEDED);
+   }
+
+   @Test(groups = "live", dependsOnMethods = "testCreate")
+   public void testListDeployments() {
+      List<Deployment> deployments = api().list();
+      assertTrue(deployments.size() > 0 && deployments.get(0).name().equals(deploymentName));
+   }
+
+   @Test(groups = "live", dependsOnMethods = {"testGetDeployment", "testListDeployments"}, alwaysRun = true)
+   public void testDelete() throws Exception {
+      List<Deployment> deployments = api().list();
+      for (Deployment d : deployments) {
+         if (d.name().contains("jcdep")) {
+            URI uri = api().delete(d.name());
+            assertNotNull(uri);
+            assertTrue(uri.toString().contains("api-version"));
+            assertTrue(uri.toString().contains("operationresults"));
+
+            boolean jobDone = Predicates2.retry(new Predicate<URI>() {
+               @Override
+               public boolean apply(URI uri) {
+                  System.out.println(uri.toString());
+                  System.out.println(api.getJobApi().jobStatus(uri).name());
+                  return ParseJobStatus.JobStatus.NO_CONTENT == api.getJobApi().jobStatus(uri);
+               }
+            }, 60 * maxTestDuration * 1000).apply(uri);
+            assertTrue(jobDone, "delete operation did not complete in the configured timeout");
+         }
+      }
+   }
+
+   private DeploymentApi api() {
+      return api.getDeploymentApi(resourceName);
+   }
+}
```

Comment:
Add a test (or call it in an existing one) that exercises the `DeploymentApi.validate` method